File: VB/DragDropDateNavigator/Form1.vb
Imports System
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Calendar
Imports DevExpress.XtraScheduler
Imports DevExpress.XtraScheduler.Drawing

Namespace XtraSchedulerTest
    Partial Public Class Form1
        Inherits Form

        Private DateNavigatorHoverFirstDate As Date = Date.MinValue

        Public Sub New()
            InitializeComponent()

            'Create resources
            For i As Integer = 0 To 3
                Dim r As Resource = Me.schedulerStorage1.CreateResource(i, String.Format("Resource {0}", i))
                schedulerStorage1.Resources.Add(r)

                For i1 As Integer = 0 To 2
                    Dim apt As Appointment = Me.schedulerStorage1.CreateAppointment(AppointmentType.Normal, Date.Today.AddDays(i).AddHours(i + 0), Date.Today.AddDays(i).AddHours(i + 1), String.Format("Appointment {0} {1}", i, i1))
                    apt.ResourceId = r.Id
                    schedulerStorage1.Appointments.Add(apt)
                Next i1
            Next i

            schedulerControl1.Start = Date.Now
            AddHandler schedulerControl1.AppointmentDrag, AddressOf schedulerControl1_AppointmentDrag
            schedulerControl1.GroupType = SchedulerGroupType.None
        End Sub


        Private Sub schedulerControl1_AppointmentDrag(ByVal sender As Object, ByVal e As AppointmentDragEventArgs) Handles schedulerControl1.AppointmentDrag
            e.ForceUpdateFromStorage = True
        End Sub

        Private Sub dateNavigator1_DragOver(ByVal sender As Object, ByVal e As DragEventArgs) Handles dateNavigator1.DragOver
            If schedulerControl1.ActiveView.Type <> SchedulerViewType.Day Then
                Return
            End If

            Dim cellDate As Date = GetDateNavigatorCellDateTimeFromPoint(New Point(e.X, e.Y))
            If cellDate <> Date.MinValue AndAlso (Not schedulerControl1.ActiveView.GetVisibleIntervals().Contains(New TimeInterval(cellDate, TimeSpan.FromHours(24)))) Then
                If DateNavigatorHoverFirstDate = Date.MinValue Then
                    DateNavigatorHoverFirstDate = cellDate
                    timer1.Start()
                ElseIf DateNavigatorHoverFirstDate <> cellDate Then
                    timer1.Stop()
                    DateNavigatorHoverFirstDate = cellDate
                    timer1.Start()
                End If
            End If
        End Sub

        Private Function GetDateNavigatorCellDateTimeFromPoint(ByVal p As Point) As Date
            Dim pt As Point = Point.Empty
            Me.Invoke(New MethodInvoker(Sub() pt = dateNavigator1.PointToClient(p)))

            Dim hitInfo As CalendarHitInfo = dateNavigator1.GetHitInfo(New MouseEventArgs(MouseButtons.None, 0, pt.X, pt.Y, 0))
            If hitInfo.HitTest = CalendarHitInfoType.MonthNumber Then
                Dim cell As CalendarCellViewInfo = TryCast(hitInfo.HitObject, CalendarCellViewInfo)
                Return cell.Date
            End If

            Return Date.MinValue
        End Function

        Private Sub timer1_Tick(ByVal sender As Object, ByVal e As EventArgs) Handles timer1.Tick

            timer1.Stop()

            Dim cellDate As Date = GetDateNavigatorCellDateTimeFromPoint(MousePosition)

            If DateNavigatorHoverFirstDate = cellDate Then
                Me.Invoke(New MethodInvoker(Sub() schedulerControl1.GoToDate(DateNavigatorHoverFirstDate)))
            End If

            DateNavigatorHoverFirstDate = Date.MinValue
        End Sub
    End Class
End Namespace

File: CS/DragDropDateNavigator/Form1.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraEditors.Calendar;
using DevExpress.XtraScheduler;
using DevExpress.XtraScheduler.Drawing;

namespace XtraSchedulerTest
{
    public partial class Form1 : Form
    {
        DateTime DateNavigatorHoverFirstDate = DateTime.MinValue;

        public Form1() {
            InitializeComponent();

            //Create resources
            for (int i = 0; i < 4; i++) {
                Resource r = new Resource(i, string.Format("Resource {0}", i));
                schedulerStorage1.Resources.Add(r);

                for (int i1 = 0; i1 < 3; i1++) {
                    Appointment apt = new Appointment(AppointmentType.Normal,
                        DateTime.Today.AddDays(i).AddHours(i + 0), DateTime.Today.AddDays(i).AddHours(i + 1),
                        string.Format("Appointment {0} {1}", i, i1));
                    apt.ResourceId = r.Id;
                    schedulerStorage1.Appointments.Add(apt);
                }
            }

            schedulerControl1.Start = DateTime.Now;
            schedulerControl1.AppointmentDrag += new AppointmentDragEventHandler(schedulerControl1_AppointmentDrag);
            schedulerControl1.GroupType = SchedulerGroupType.None;
        }


        void schedulerControl1_AppointmentDrag(object sender, AppointmentDragEventArgs e) {
            e.ForceUpdateFromStorage = true;
        }
     
        private void dateNavigator1_DragOver(object sender, DragEventArgs e) {
            if (schedulerControl1.ActiveView.Type != SchedulerViewType.Day)
                return;           

            DateTime cellDate = GetDateNavigatorCellDateTimeFromPoint(new Point(e.X, e.Y));
            if (cellDate != DateTime.MinValue &&
                !schedulerControl1.ActiveView.GetVisibleIntervals().Contains(new TimeInterval(cellDate, TimeSpan.FromHours(24))))
            {
                if (DateNavigatorHoverFirstDate == DateTime.MinValue)
                {
                    DateNavigatorHoverFirstDate = cellDate;
                    timer1.Start();
                }
                else if (DateNavigatorHoverFirstDate != cellDate)
                {
                    timer1.Stop();
                    DateNavigatorHoverFirstDate = cellDate;
                    timer1.Start();
                }
            }
        }

        private DateTime GetDateNavigatorCellDateTimeFromPoint(Point p)
        {
            Point pt = Point.Empty;
            this.Invoke(new MethodInvoker(delegate()
            {
                pt = dateNavigator1.PointToClient(p);
            }));

            CalendarHitInfo hitInfo = dateNavigator1.GetHitInfo(new MouseEventArgs(MouseButtons.None, 0, pt.X, pt.Y, 0));
            if (hitInfo.InfoType == CalendarHitInfoType.MonthNumber)
            {
                return ((DateNavigatorDayNumberCellInfo)hitInfo.HitObject).Date;
            }

            return DateTime.MinValue;
        }

        private void timer1_Tick(object sender, EventArgs e)
        {
  
            timer1.Stop();

            DateTime cellDate = GetDateNavigatorCellDateTimeFromPoint(MousePosition);

            if (DateNavigatorHoverFirstDate == cellDate)
            {                
                this.Invoke(new MethodInvoker(delegate()
                {
                    schedulerControl1.GoToDate(DateNavigatorHoverFirstDate);             
                }));
            }

            DateNavigatorHoverFirstDate = DateTime.MinValue;
        } 
    }
}